@0xc0170 

the failure is because:

```
[Error] itm_api.c@34,30: too few arguments to function call, expected 2, have 0
[ERROR] ./targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/itm_api.c:34:30: error: too few arguments to function call, expected 2, have 0
    am_bsp_itm_printf_enable();
    ~~~~~~~~~~~~~~~~~~~~~~~~ ^
./targets/TARGET_Ambiq_Micro/TARGET_Apollo3/TARGET_SFE_ARTEMIS_MODULE/bsp/am_bsp.h:162:13: note: 'am_bsp_itm_printf_enable' declared here
extern void am_bsp_itm_printf_enable(uint32_t ui32Pin, am_hal_gpio_pincfg_t sPincfg);
            ^
1 error generated.
```

As am_bsp_itm_printf_enable can take 2 options 

```
#ifdef AM_BSP_GPIO_ITM_SWO
extern void am_bsp_itm_printf_enable(void);
#else
extern void am_bsp_itm_printf_enable(uint32_t ui32Pin, am_hal_gpio_pincfg_t sPincfg);
#endif
```

in the same time `itm_init` take no arguments. 

How I should keep things compatible without changing the BSP for each Apollo-based target?

--- a/targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/itm_api.c
+++ b/targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/itm_api.c
@@ -25,13 +25,24 @@
 
 /* SWO frequency: 1000 kHz */
 
-// As SWO has to be accessible everywhere, including ISRs, we can't easily
-// communicate the dependency on clocks etc. to other components - so this
-// function checks that things appear to be set up, and if not re-configures
-// everything
+#ifndef AM_BSP_GPIO_ITM_SWO
+#define AM_GPIO_ITM_SWO             	22
+
+const am_hal_gpio_pincfg_t g_AM_GPIO_ITM_SWO =
+{
+    .uFuncSel            = AM_HAL_PIN_22_SWO,
+    .eDriveStrength      = AM_HAL_GPIO_PIN_DRIVESTRENGTH_2MA
+};
+#endif 
+
+
 void itm_init(void)
 {
+    #ifdef AM_BSP_GPIO_ITM_SWO
     am_bsp_itm_printf_enable();
+    #else
+    am_bsp_itm_printf_enable(AM_GPIO_ITM_SWO,g_AM_GPIO_ITM_SWO);
+    #endif
 }
 
 #endif

